### ISSUE TITLE
chromiumedgedriver 404's during selenium-standalone install (v8.0.4)

### ISSUE THREAD

**Issue by third774** [webdriverio/selenium-standalone#585]
## Expected Behavior
Expecting to be able to run these commands in succession and have selenium start up:
```sh
yarn init -y
yarn add selenium-standalone
npx selenium-standalone install
npx selenium-standalone start
```

## Current Behavior
It seems that the `chromiumedgedriver` fails to download because of a 404

```
╭─ kevin  ~/experiments/selenium-standalone-issue  
╰ ﬦ npx selenium-standalone install                                                                                                                                                                                         ✔  10092  11.11.21  10:38:13
----------
selenium-standalone installation starting
----------

---
selenium install:
from: https://github.com/SeleniumHQ/selenium/releases/download/selenium-4.0.0/selenium-server-4.0.0.jar
to: /Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/.selenium/selenium-server/4.0.0/selenium-server.jar
---
chrome install:
from: https://chromedriver.storage.googleapis.com/96.0.4664.45/chromedriver_mac64.zip
to: /Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/.selenium/chromedriver/latest-x64/chromedriver
---
firefox install:
from: https://github.com/mozilla/geckodriver/releases/download/v0.30.0/geckodriver-v0.30.0-macos.tar.gz
to: /Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/.selenium/geckodriver/latest-x64/geckodriver
---
chromiumedge install:
from: https://msedgedriver.azureedge.net/96.0.1054.29/edgedriver_mac64.zip
to: /Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/.selenium/chromiumedgedriver/latest-x64/msedgedriver

[===                 ] 13%      https://chromedriver.storage.googleapis.com/96.0.4664.45/chromedriver_mac64.zipError in "getDownloadStream". Could not download https://msedgedriver.azureedge.net/96.0.1054.29/edgedriver_mac64.zip
See more details below:
404 https://msedgedriver.azureedge.net/96.0.1054.29/edgedriver_mac64.zip
Response code 404 (Not Found)
[====================] 100%      https://chromedriver.storage.googleapis.com/96.0.4664.45/chromedriver_mac64.zip
[====================] 100%      https://github.com/SeleniumHQ/selenium/releases/download/selenium-4.0.0/selenium-server-4.0.0.jar


╭─ kevin  ~/experiments/selenium-standalone-issue  
╰ ﬦ npx selenium-standalone start                                                                                                                                                                                           ✔  10093  11.11.21  10:38:23
/Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/lib/check-paths-existence.js:13
        throw new Error('Missing ' + path);
              ^

Error: Missing /Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/.selenium/chromiumedgedriver/latest-x64/msedgedriver
    at /Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/lib/check-paths-existence.js:13:15
    at async Promise.all (index 2)
    at async Object.start (/Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/lib/start.js:113:3)
    at async Object.start (/Users/kevin/experiments/selenium-standalone-issue/node_modules/selenium-standalone/bin/selenium-standalone:23:16)
    ```

## Steps to Reproduce
Provide an unambiguous set of steps to reproduce this bug. Include commands that you used
in the shell, or snippets of how you are using the JavaScript API.

1.  Create an empty folder to test in
2.  Run the following command:

```sh
yarn init -y
yarn add selenium-standalone
npx selenium-standalone install
npx selenium-standalone start
```

## Your Environment
* Version of `selenium-standalone` that you are using: 8.0.4
* Is there another tool calling `selenium-standalone` on your behalf: Nope (unless you count `npx`)
* System/platform: macOS 11.6.1 (Big Sur), Intel chip

**Comment by nathanKramer** [comment]
This seems like the beginning of a workaround, I'm still having trouble starting `selenium-standalone` though

```shell
cd node_modules/selenium-standalone/.selenium/chromiumedgedriver/latest-x64
curl -0  --output msedgedriver.zip
unzip msedgedriver.zip msedgedriver
```

**Comment by harsha509** [comment]
msedgedriver  'Version: 96.0.1054.29' is not available for Mac and Linux. you can check it in here https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/,

but still, WDIO tries to download it. Maybe you can try replacing other versions like 'Version: 96.0.1054.34' in the config file.

(or) WDIO needs to check before download in `getDownloadStream`

**Comment by christian-bromann** [comment]
@harsha509 should wdio just silently fail with a not that the browser is not available for the given platform?

**Comment by harsha509** [comment]
Hi @christian-bromann,

* Can we download only available drivers and print a warning message that the browser driver is not available for a given platform, and start the server (This only works if the browser declared is x and driver failed to download is Y)

(or)

* Can we fail silently and print messages like `use drivers: {chrome: true} to avoid downloading other drivers`

(or)

* Can we fail silently and print a warning message that the browser driver is not available for a given platform, try a different driver version available from [here ](https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/)
(Again this enforces the user to specify the driver version in the config file)

In my view, this is an issue from the driver provider and not from WebDriverIO, So printing a precise message in the console will do!

Please suggest on this, I'll work on this.

Thanks,
Sri Harsha

**Comment by christian-bromann** [comment]
> So printing a precise message in the console will do!

Totally agree with this. Let's implement it this way.

**Comment by nathanKramer** [comment]
For those using `wdio`, you can use this in your config to download / use specific drivers:

```
services: [
    [
      'selenium-standalone',
      {
        installArgs: {
          drivers: {
            chrome: true,
          },
        },
        args: {
          drivers: {
            chrome: true,
          },
        },
      },
    ],
```

**Comment by lthurr** [comment]
I believe this is solved by #595 
The behaviour now is that it will check if the bundle is available to download, if it is not, then it will download the fallback version

**Comment by christian-bromann** [comment]
Closing then. Please let us know if issues still occur, then we can re-open.
